lib/gruf/interceptors/instrumentation/output_metadata_timer.rb

```diff
@@ -15,29 +15,34 @@
 # OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #
 module Gruf
-  module Instrumentation
-    module RequestLogging
-      module Formatters
+  module Interceptors
+    module Instrumentation
+      ##
+      # Appends the timer metadata to the active call output metadata
+      #
+      class OutputMetadataTimer < ::Gruf::Interceptors::ServerInterceptor
         ##
-        # Formats the request into plaintext logging
+        # Handle the instrumented response. Note: this will only instrument timings of _successful_ responses.
         #
-        class Plain < Base
-          ##
-          # Format the request by only outputting the message body and params (if set to log params)
-          #
-          # @param [Hash] payload The incoming request payload
-          # @return [String] The formatted string
-          #
-          def format(payload)
-            time = payload.fetch(:duration, 0)
-            grpc_status = payload.fetch(:grpc_status, 'GRPC::Ok')
-            route_key = payload.fetch(:method, 'unknown')
-            body = payload.fetch(:message, '')
+        def call
+          return unless request.active_call && request.active_call.respond_to?(:output_metadata)
```

Comment:
Since `nil` responds to `respond_to?`, you can get away with just `request.active_call.respond_to?(:output_metadata)` here.